### DISCUSSION TITLE
Where to put shinyOptions in a golem project?

### DISCUSSION THREAD

**Discussion by zilch42** [ThinkR-open/golem#892] [Q&A] [answered]
Hi there,

I am wondering where to put the following code in my golem app? Publishing to RStudio Connect.

```
shinyOptions(cache = cachem::cache_disk("./app_cache/cache/"))
```

The shiny caching documentation says to put this at the top of your app.R, server.R, or global.R. I currently have it at the top of my app_server.R file (outside of my app_server function) and that works but it just seems clunky and out of place. Everything else seems to have a place in golem. 

Any advice would be appreciated, thanks.

**Comment by ColinFay** [comment, marked as answer]
Hey, 

There are several ways to do that, and they depend on how you want to deploy your app:

- using an .Rprofile 
- for dev purpose, adding it to dev/run_dev.R (in the spirit of https://github.com/ThinkR-open/golem/blob/master/inst/shinyexample/dev/run_dev.R#L5) 
- If you deploy on RStudio product, add it to the app.R created when running one of the golem::add_rstudio*() functions 
- If you deploy with Docker, either set an .Rprofile, or set the option inside the CMD https://github.com/ColinFay/hexmake/blob/master/Dockerfile#L41 
- If you plan on releasing on a CRAN like repo, I would suggest using a golem_option instead https://github.com/ThinkR-open/golem/blob/master/R/with_opt.R#L69

I generally go for setting the options in run_dev.R then in app.R / in the Dockerfile, because it allows me to set different options in dev and prod.

**Reply by zilch42** [reply]
Awesome thank you. That looks cleaner and works for me on RStudio Connect. The different options for production and development is definitely handy too
